### PR TITLE
[codestyle] replaced unnecessary loop with clear method

### PR DESCRIPTION
**Pull request by akshitdewan**

### CODE REVIEW

**Comment by akshitdewan** [comment]
fixed

File: xwiki-rendering-macros/xwiki-rendering-macro-footnotes/src/main/java/org/xwiki/rendering/internal/macro/footnote/PutFootnotesMacro.java [hunk 1]
```diff
@@ -175,9 +175,8 @@ private ListItemBlock processFootnote(MacroMarkerBlock footnoteMacro, int counte
     private void addFootnoteRef(MacroMarkerBlock footnoteMacro, Block footnoteRef)
     {
         for (ListIterator<Block> it = footnoteMacro.getChildren().listIterator(); it.hasNext();) {
```
Comment (tmortagne) [inline xwiki-rendering-macros/xwiki-rendering-macro-footnotes/src/main/java/org/xwiki/rendering/internal/macro/footnote/PutFootnotesMacro.java]: Seems to be this whole code is equivalent to `footnoteMacro.getChildren().clear();`

**Comment by akshitdewan** [comment]
Fixed

**Comment by tmortagne** [comment]
Still this whole method does not make much sense. It's documented as adding something when in fact it's replacing everything.

Let's say the good side if that your change makes it very clear how weird this method is :)

**Comment by tmortagne** [comment]
Thanks for your contribution !

**Pull request merged**